### PR TITLE
JSON-B usage example

### PR DESCRIPTION
**Pull request by Verdent**
JSON-B example added. I have also changed version of javaee-api from 7.0 to 8.0-b22. Is that ok to do that this way? I think that adding separate dependency for JSON-B was not the correct way to go.

### CODE REVIEW

File: web/jsonb/jsonbbasics/src/main/java/javaeetutorial/jsonbmodel/JsonbBean.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+import java.io.*;
+import java.util.*;
+import javax.enterprise.context.SessionScoped;
+import javax.inject.Named;
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+import javax.json.bind.JsonbConfig;
+
+/* This class manages the data from a JSF form, creates
```
Comment (m0mus) [inline web/jsonb/jsonbbasics/src/main/java/javaeetutorial/jsonbmodel/JsonbBean.java]: I see some inconsistency in comments format. Other files are using different style of comments. Please make comments in this file the same as in others. This is a sample and it should look perfect from the formatting perspective.

File: web/jsonb/jsonbbasics/src/main/java/javaeetutorial/jsonbmodel/JsonbBean.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+    protected String jsonTextArea = "";
+    
+    public JsonbBean() {}
+    
+    /* Getters and setters */
+    public String getPhoneTypeHome() {
+        return PHONE_TYPE_HOME;
+    }
+    public String getPhoneTypeWork() {
```
Comment (m0mus) [inline web/jsonb/jsonbbasics/src/main/java/javaeetutorial/jsonbmodel/JsonbBean.java]: I would add an empty line between methods.

File: web/jsonb/jsonbbasics/src/main/java/javaeetutorial/jsonbmodel/JsonbBean.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        /* JSON-B config and Jsonb instance creation */
+        JsonbConfig config = new JsonbConfig()
+                .withFormatting(true);
+        Jsonb jsonb = JsonbBuilder.create(config);
+
+        /* Serialization to JSON */
+        this.jsonTextArea = jsonb.toJson(person);
+        
+        /* JSF navigation */
+        return "jsongenerated";
```
Comment (m0mus) [inline web/jsonb/jsonbbasics/src/main/java/javaeetutorial/jsonbmodel/JsonbBean.java:135]: Should it be a constant? I am not sure about it, not familiar with JSF best practices. :)

File: web/jsonb/jsonbbasics/src/main/java/javaeetutorial/jsonbmodel/JsonbBean.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    }
+    
+    /* Action method for form in jsongenerated.xhtml.
+     * Deserialize JSON from the textarea to Person object
+     * and fills these data to form. */
+    public String parseJson() {
+        clearFields();
+
+        /* JSON-B config and Jsonb instance creation */
+        Jsonb jsonb = JsonbBuilder.create(new JsonbConfig());
```
Comment (m0mus) [inline web/jsonb/jsonbbasics/src/main/java/javaeetutorial/jsonbmodel/JsonbBean.java]: Is it possible to create jsonb engine once per the application lifecycle? Creating it on every request is not the best practice.

File: web/jsonb/jsonbbasics/src/main/java/javaeetutorial/jsonbmodel/JsonbBean.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                    this.phoneType2 = number.getType();
+                } else {
+                    break;
+                }
+                index++;
+            }
+        }
+
+        /* JSF navigation */
+        return "index";
```
Comment (m0mus) [inline web/jsonb/jsonbbasics/src/main/java/javaeetutorial/jsonbmodel/JsonbBean.java:167]: Should it be a constant?

File: web/jsonb/jsonbbasics/src/main/java/javaeetutorial/jsonbmodel/JsonbBean.java (hunk 3 above)
Comment (Verdent) [reply web/jsonb/jsonbbasics/src/main/java/javaeetutorial/jsonbmodel/JsonbBean.java:135]: To use it as constant like this, is one of the possible navigation in JSF.

**Comment by m0mus** [comment]
Looks good to me.

**Pull request merged**